src/driver/linux_onload/onload_kernel_compat.h

```diff
@@ -121,11 +121,15 @@ static inline int efrm_unregister_netdevice_notifier(struct notifier_block *b)
 #ifndef EFRM_HAS_LOOKUP_FDGET_RCU
 static inline struct file *lookup_fdget_rcu(unsigned int fd)
 {
+  struct file* f;
+
 #ifdef EFRM_HAS_LOOKUP_FD_RCU
-  return lookup_fd_rcu(fd);
+  f = lookup_fd_rcu(fd);
 #else
-  return fcheck(fd);
+  f = fcheck(fd);
 #endif
+
+  return f ? get_file(f) : NULL;
```

Comment:
I do not understand.  In the case of "new" kernel, are we taking 2 refcounts, one in `lookup_fd_rcu`, and another in `get_file`?

Reply:
How about if we implement a function something like `bool oo_check_file(int fd, struct file *filp)` that would either call `lookup_fd_rcu()` or `fcheck()` and then call `fput()` in the former case, so that the caller would not need to decrement the reference counter?

The `get_file()` seems excessive in our case.

Reply:
> I do not understand. In the case of "new" kernel, are we taking 2 refcounts, one in `lookup_fd_rcu`, and another in `get_file`?

With "new" kernel we use kernel version of `lookup_fdget_rcu()`, which increments refcount.
With "old" kernels we use this wrapper, that does `lookup_fd_rcu()` + `get_file()`. 
`lookup_fd_rcu()` does not increment refcount.

Reply:
> How about if we implement a function something like `bool oo_check_file(int fd, struct file *filp)` that would either call `lookup_fd_rcu()` or `fcheck()`, compare the pointers and then call `fput()` in the former case, so that the caller would not need to decrement the reference counter?
> 
> The `get_file()` seems excessive in our case.

Good proposal. I'll rewrite it that way.

Reply:
Got it, thank you!

Reply:
Ah, now I get it too!

There are three functions:
- `fcheck`
- `lookup_fd_rcu`
- `lookup_fdget_rcu`

The first two (used in Onload) do not bump the reference counter. The last one does, but Onload does not use it.

It means that as-is this PR does not fix anything because there is no resource leak, but still there is a possibility for us to handle all these three and call `fput` to cope with `lookup_fdget_rcu` which is the only available of these three in the latest kernel, right? In this case, the `oo_check_file()` suggestion still sounds adequate?

Reply:
Yes, in linux-6.7 we only have `lookup_fdget_rcu()`. After calling it we must do `fput()`.
In older kernels we have `lookup_fd_rcu`, in the oldest - `fcheck`.
I created a wrapper with the latest name and make it increment refcount also, to mimic new behavior. In such case the caller uses the latest interface regardless of kernel version.

We may implement `oo_check_file()` and deal with refcounting and compatibilty there.
For me both variants look reasonable, but I'd keep current solution.